### PR TITLE
PIM-10345: Having a full numerical attribute code cause an error on product model import

### PR DESCRIPTION
**Pull request by MadeWilson**
In this PR: 

I fixed the following error : When user import a product with a numeric only attribute name, we have a fatal error with the following message: `preg_match(): #2 ($subject) must be of type string, int given.`

**Why we didn't have this error message in product import ?** 
Because src/Akeneo/Pim/Enrichment/Component/Product/Connector/ArrayConverter/FlatToStandard/Product.php doesn't have `declare(strict_types=1);` so the cast is automatically done by PHP.

Current PHPStan level is 2 on enrichment, only level 5 can catch this error. 

Instead of removing the `declare(strict_types=1);` i manually casted the variable

### CODE REVIEW

**Review by SamirBoulil — APPROVED** [review]

File: tests/back/Pim/Enrichment/Specification/Component/Product/Connector/ArrayConverter/FlatToStandard/ProductModelSpec.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace Specification\Akeneo\Pim\Enrichment\Component\Product\Connector\ArrayConverter\FlatToStandard;
 
+use Akeneo\Tool\Component\Connector\Exception\ArrayConversionException;
```
Comment (ghoussard) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Connector/ArrayConverter/FlatToStandard/ProductModelSpec.php:5]: Unused import

**Review by ghoussard — APPROVED** [review]
Don't forget the CHANGELOG

**Comment by BitOne** [comment]
Your description is missing.

It would be necessary to have more insight on when this variable became an integer. Here you're treating the symptoms (checking if it's an integer and cast it to string). We must fix the root cause before anything. And only resorting on treating the symptoms when we don't have the choice.

So an analysis of the why is required here in your description.

Moreover, can you add more tests please, in order to cover a wider range of risks, please?

For example:
 - float numbers
 - string beginning with a number
 - real string but with only numbers in it (here you have an int)
 - real string with a float in them
 
 But the tests should really be done where the root issue is really.

**Review by BitOne — COMMENTED** [review]
See comment above.

**Pull request merged**